networkx/algorithms/isomorphism/ismags.py

```diff
@@ -89,12 +89,158 @@
+  side of the DFS. They are separated from other nodes during the partition refinement
+  process after coupling. So they never get coupled directly. Thus the number of cosets
+  we find if typically many fewer than the number of nodes.
+
+  We turn those cosets into constraints on the nodes when building non-symmetric
+  isomorphisms. The node who's coset is used is paired with each other node in the
+  coset. These node-pairs form the constraints. During isomorphism construction we
+  always select the first of the constraint before the other. This removes subtrees
+  from the DFS traversal space used to build isomorphisms.
+
+  The constraints we obtain via symmetry analysis of the subgraph are used for
+  finding non-symmetric isomorphisms. We prune the isomorphism tree based on
+  the constraints we obtain from the symmetry analysis.
+
+  **Isomorphism Construction**
+
+  Once we have symmetry constraints on the isomorphisms, ISMAGS constructs the allowed
+  isomorphisms by mapping each node of the subgraph to all possible nodes (with the
+  same degree-by-color) from the graph. We partition all nodes into degree-by-color
+  parts and order the subgraph nodes we consider using smallest part size first.
+  The idea is to try to map the most difficult subgraph nodes first (most difficult
+  here means least number of graph candidates).
+
+  By considering each potential subgraph node to graph candidate mapping image in turn,
+  we perform a DFS traversal of partial mappings. If the mapping is rejected due to
+  the graph neighbors not matching the degree-by-color of the subgraph neighbors, or
+  rejected due to the constraints imposed from symmetry, we prune that subtree and
+  consider a new graph candidate node for that subgraph node. When no more graph
+  candidates remain we backtrack to the previous node in the mapping and consider a
+  new graph candidate for that node. Is we ever to a depth where all subgraph nodes
+  are mapped and no structural requirements or symmetry constraints are violated,
+  we have found an isomorphism. We yield that mapping and backtrack to find other
+  isomorphisms.
+
+  As we visit more neighbors, the graph candidate nodes have to satisfy more structural
+  restrictions. As described in the ISMAGS paper, [1]_, we store each set of structural
+  restrictions separately as a set of possible candidate nodes rather than computing
+  the intersection of that set with the known graph candidates for the subgraph node.
+  We delay taking the intersection until that node is selected to be in the mapping.
+  While choosing the node with fewest candidates, we avoid computing the intersection
+  by using the size of the minimal set to be intersected rather than the size of the
+  intersection. This may make the node ordering slightly worse via a savings of
+  many intersections most of which are not ever needed.
+
+- With a method that yields subgraph_isomorphisms, we can construct functions like
```

Comment:
Related to the comment above re: the bulleted list formatting, perhaps move this one above the big explanation!